app/views/admin/versions/_object_desc_and_link.html.haml

```diff
@@ -1,156 +1,161 @@
 
-- elsif version.item_type == 'Subscription' || version.item_type == 'Registration'
+- when 'Subscription', 'Registration'
   = 'conference'
   = link_to Conference.find(version.conference_id).title,
           admin_conference_registrations_path(conference_id: Conference.find(version.conference_id).short_title)
 
-- elsif version.item_type == 'Commercial'
-  - commercial_last_version = get_version_object(PaperTrail::Version.where(item_type: version.item_type, item_id: version.item_id).last)
-  - commercialable_last_version = get_version_object(PaperTrail::Version.where(item_type: commercial_last_version.commercialable_type,
-                                                          item_id: commercial_last_version.commercialable_id).last)
+- when 'Commercial'
+  - commercial = current_or_last_object_state(version.item_type, version.item_id)
+  - commercialable = current_or_last_object_state(commercial.commercialable_type, commercial.commercialable_id)
 
-  - case commercial_last_version.commercialable_type
+  - case commercial.commercialable_type
   - when 'Event'
-    =  link_to 'commercial',
-            edit_admin_conference_program_event_path(conference_id: Conference.find(version.conference_id).short_title,
-                id: commercialable_last_version.id, anchor: 'commercials-content')
-    = "in event #{commercialable_last_version.title}"
+    commercial in event
+    =  link_to commercialable.title,
+           admin_conference_program_event_path(conference_id: Conference.find(version.conference_id).short_title, id: commercialable.id)
 
   - when 'Venue'
-    - if Venue.find_by(id: commercialable_last_version.id)
-      =  link_to 'commercial',
-              edit_admin_conference_program_event_path(conference_id: Conference.find(version.conference_id).short_title,
-                  id: commercialable_last_version.id, anchor: 'commercials-content')
-    - else
-      = 'commercial'
-    = "in venue #{commercialable_last_version.name}"
+    commercial in venue
+    = link_to commercialable.name,
+          edit_admin_conference_venue_path(conference_id: Conference.find(version.conference_id).short_title,
+                                              id: commercialable.id, anchor: 'commercials-content')
 
   - when 'Conference'
     = link_to 'commercial',
             admin_conference_commercials_path(conference_id: Conference.find(version.conference_id).short_title)
 
-- elsif %w(EventsRegistration Comment Vote).include?(version.item_type)
+- when 'EventsRegistration', 'Comment', 'Vote', 'Event'
   = 'event'
-  - event_id = get_version_object(version).try(:event_id) || get_version_object(version).try(:commentable_id)
-  = link_to Event.find(event_id).title,
+  - object = current_or_last_object_state(version.item_type, version.item_id)
+  - event_id = object.try(:id) || object.try(:event_id) || object.try(:commentable_id)
+  = link_to current_or_last_object_state('Event', event_id).title,
```

Comment:
I didn't understand your comment.

To make it clear, the reason why its assumed here that there will be event records is because the Event model was already being tracked, even before #1019.They would be fetched properly by current_or_last_object_state even if the rake task for adding conference_id has not been run as `current_or_last_object_state` uses model_name and id